classes/merge_request.php

```diff
@@ -0,0 +1,159 @@
+     * @var integer
+     */
+    const COMPLETED_WITH_ERRORS = 7;
+    /**
+     * Reference table for merge requests.
+     * @var string
+     */
+    const TABLE_MERGE_REQUEST = 'tool_mergeusers_queue';
+     /**
+     * Reference table for merge requests. Before web service integration.
+     * @var string
+     */
+    const TABLE_MERGE_REQUEST_OLD = 'tool_mergeusers';
+    /**
+     * Reference table for users.
+     * @var string
+     */
+    const TABLE_USERS = 'user';
+    private $data;
+    private function __construct(stdClass $data) {
+        $this->data = $data;
+    }
+
+    public static function from(stdClass $data) {
+        return new self($data);
+    }
+
+    public function __get($name) {
+        if (isset($this->data->${name})) {
+            $value = $this->data->${name};
+            if ($name == 'log') {
+                $value = json_decode($value, true);
+            }
+            return $value;
+        }
+        return null;
+    }
+    public function __set($name, $value) {
+        if (isset($this->data->${name})) {
+            if ($name == 'log') {
+                $value = json_encode($value);
+            }
+            $this->data->${name} = $value;
+        }
+    }
+
+    public function convert_log_to_new_format() {
+        if (!isset($this->data->log) || empty($this->data->log)) {
+            return;
+        }
+        $logs = [];
+        $logs[1] = json_decode($this->data->log, true);
+        $this->data->log = json_encode($logs);
+    }
+
+    public function get_record(): \stdClass {
+        return $this->data;
+    }
+
+    public static function export_data_to_new_table(): object {
+        global $DB;
+        $filter = array('status' => self::QUEUED_NOT_PROCESSED);
+        $sort = "id DESC";
+        $fields = "id, touserid, fromuserid, success, timemodified, log";
+        $records = $DB->get_recordset(self::TABLE_MERGE_REQUEST_OLD,
+                                    $filter, $sort, $fields);
+        if (!$records->valid()) {
+            return $records;
+        }
+        foreach ($records as $item) {
+            // Insert item into new table.
+            if ($item->status == 1) {
+                $status = self::COMPLETED_WITH_SUCCESS;
+            } else if ($item->status == 0) {
+                $status = self::COMPLETED_WITH_ERRORS;
+            }
+            $logs = [];
+            // Take all data of $item->fromuserid and $item->touserid.
+            $logs[1] = json_decode($item->log, true);
+            $idrecord = $DB->insert_record(
+                self::TABLE_MERGE_REQUEST ,
+                [
+                    'removeuserid' => $item->fromuserid,
+                    'removeuserfield' => 'id',
+                    'removeuservalue' => $item->fromuserid,
+                    'keepuserid' => $item->touserid,
+                    'keepuserfield' => 'id',
+                    'keepuservalue' => $item->touserid,
+                    'timeadded' => time(),
```

Comment:
instead of using `time()` for `timeadded` and `timemodified`, use the `$item->timemodified`. Do not rewrite the history.